### ISSUE TITLE
firebase_messaging not working if include flutter_local_notifications package on iOS

### ISSUE THREAD

**Issue by ishakdas** [FirebaseExtended/flutterfire#1581]
Hello to everyone
I was able to send data notification in the background via android but I think there is a problem for ios.
Is this Problem a solved issue or is it still going on?

**Comment by iapicca** [comment]
Hi @ishakdas 
[flutter_local_notifications](https://pub.dev/packages/flutter_local_notifications) is a 3rd party plugin,
if you are experiencing problems with that you may want to open an issue in the dedicated [github](https://github.com/MaikuB/flutter_local_notifications/issues).

If you are experiencing issues with [firebase_messaging ](https://pub.dev/packages/firebase_messaging) 
please provide your   `flutter doctor -v`
and your `flutter run --verbose`.
Also, to better address the issue, would be helpful
if you could post  a minimal code sample
or the steps to reproduce the problem.
Thank you

**Comment by ishakdas** [comment]
The firebase plug-in works alone but does not work with the flutter_local_notifications plug-in.
It doesn't work in the background.
The application works while it is open.

**Comment by ishakdas** [comment]
```
[√] Flutter (Channel stable, v1.9.1+hotfix.6, on Microsoft Windows [Version 10.0.18362.476], locale tr-TR)
    • Flutter version 1.9.1+hotfix.6 at C:\flutter
    • Framework revision 68587a0916 (3 months ago), 2019-09-13 19:46:58 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

 
[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\Users\admintrack\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 41.0.2
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] VS Code (version 1.40.2)
    • VS Code at C:\Users\admintrack\AppD
```

**Comment by ishakdas** [comment]
```

FlutterLocalNotificationsPlugin flutterLocalNotificationsPlugin =
    FlutterLocalNotificationsPlugin();
//deneme
Future<void> myBackgroundMessageHandler(Map<String, dynamic> message) {
  if (message.containsKey('data')) {
    NotificationHandler.showNotification(message);
  }

  if (message.containsKey('notification')) {}

  return Future<void>.value();
}
 

BuildContext myContext;

class NotificationHandler {
  
  FirebaseMessaging _fcm = FirebaseMessaging();
  static final NotificationHandler _singleton = NotificationHandler._internal();
  factory NotificationHandler() {
    return _singleton;
  }
  NotificationHandler._internal();
  initializeFCMNotification(BuildContext context) async {
   

    //print("csdsdf");
    myContext = context;
    var initializationSettingsAndroid =
        AndroidInitializationSettings('app_icon');
    var initializationSettingsIOS = IOSInitializationSettings(
        onDidReceiveLocalNotification: onDidReceiveLocalNotification);
    var initializationSettings = InitializationSettings(
        initializationSettingsAndroid, initializationSettingsIOS);
    flutterLocalNotificationsPlugin.initialize(initializationSettings,
        onSelectNotification: onSelectNotification);
    _fcm.subscribeToTopic("all");

    if (Platform.isIOS) {
      _fcm.requestNotificationPermissions(
          IosNotificationSettings(sound: true, badge: true, alert: true));
      _fcm.onIosSettingsRegistered.listen((IosNotificationSettings settings) {
        print("Settings registered: $settings");
      });
    }

    _fcm.configure(
      onMessage: (Map<String, dynamic> message) async {
       print('01 $message');
     
        showNotification(message);
      },
      onResume: (Map<String, dynamic> message) async {
        print('02  $message');
       
        showNotification(message);
      },
      onBackgroundMessage: Platform.isIOS ?null:myBackgroundMessageHandler,
      onLaunch: (Map<String, dynamic> message) async {
        print('03  $message');
        showNotification(message);
      },
    );
  }

  static void showNotification(Map<String, dynamic> message) async {
   

    var mesaj = Person(
        name:  Platform.isIOS ? message["title"]:message["data"]["title"],
        key: '1',
        iconSource: IconSource.Drawable);
        //iconSource: IconSource.FilePath);
    var mesajStyle = MessagingStyleInformation(mesaj,
        messages: [Message(Platform.isIOS ?message["body"]:message["data"]["body"], DateTime.now(), mesaj)]);

    var androidPlatformChannelSpecifics = AndroidNotificationDetails(
        '1234', 'Yeni Mesaj', 'your channel description',
        style: AndroidNotificationStyle.Messaging,
        styleInformation: mesajStyle,
        importance: Importance.Max,
        priority: Priority.High,
        playSound: true,
        enableVibration: true,
        ticker: 'ticker');
    var iOSPlatformChannelSpecifics = IOSNotificationDetails();
    var platformChannelSpecifics = NotificationDetails(
        androidPlatformChannelSpecifics, iOSPlatformChannelSpecifics);
    await flutterLocalNotificationsPlugin.show(0, Platform.isIOS ?message["title"]:message["data"]["title"],
        Platform.isIOS ?message["body"]:message["data"]["body"], platformChannelSpecifics,
        payload: jsonEncode(message));
  }

  Future onDidReceiveLocalNotification(
      int id, String title, String body, String payload) async {
    await showDialog(
      context: myContext,
      builder: (BuildContext context) => CupertinoAlertDialog(
        title: title != null ? Text(title) : null,
        content: body != null ? Text(body) : null,
        actions: [
          CupertinoDialogAction(
            isDefaultAction: true,
            child: Text('Ok'),
            onPressed: () async {
              Map<String, dynamic> gelenSoru = await jsonDecode(payload);
              await Navigator.push(
                myContext,
                MaterialPageRoute(
                    builder: (context) => BildirimDenemeSorulari(
                          url: Platform.isIOS ?gelenSoru["url"]:gelenSoru['data']['url'],
                        )),
              );
            },
          )
        ],
      ),
    );
  }
}

Future<void> onSelectNotification(String payload) async {
  if (payload != null) {
    debugPrint('notification payload: ' + payload);

    Map<String, dynamic> gelenSoru = await jsonDecode(payload);
    await Navigator.push(
      myContext,
      MaterialPageRoute(
          builder: (context) => BildirimDenemeSorulari(
                url: Platform.isIOS ?gelenSoru["url"]:gelenSoru['data']['url'],
              )),
    );
  }
}

```

**Comment by ThinkDigitalSoftware** [comment]
This is an old and ongoing issue with the local_notifications plugin. There are more details there. It has something to do with both of the plugins registering on the same channel or something. This should be closed here and you can follow the [issue](https://github.com/MaikuB/flutter_local_notifications/issues/111) on their side.

**Comment by MaikuB** [comment]
Just saw this but it wasn't an issue with the `flutter_local_notifications` plugin. Changes were made in the Flutter engine to allow multiple notification plugins to register themselves. `flutter_local_notifications` was updated to accomodate for this but the code changes for `firebase_messaging` hadn't been done. There was an issue to track that on this repository at https://github.com/FirebaseExtended/flutterfire/issues/1455. The PR (https://github.com/FirebaseExtended/flutterfire/pull/1829) I created to update the `firebase_messaging` plugin to resolve this issue was just recently merged in and now needs the new version of the plugin to be released
